spec/cohortManager.spec.js

```diff
@@ -0,0 +1,202 @@
+    const result = cohortManager.search(6)
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('returns a error since it is more than capacity', () => {
+    cohort5.students = ['studentA', 'studentB', 'studentC']
+    cohort5.add(arisaSigrist)
+    cohort5.add(bobRoss)
+
+    const expected = cohort5.add(michelleObama)
+    // evaluate
+    const result = Error('this cohort hit the capacity')
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('removes specific cohort and returns the cohort list', () => {
+    //   setup
+    cohortManager.add(cohort3)
+    cohortManager.add(cohort4)
+    cohortManager.add(cohort5)
+
+    const expected = [cohort3, cohort5]
+    // evaluate
+    const result = cohortManager.remove(4)
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('returns an error for trying to remove a non-existing cohort', () => {
+    //   setup
+    cohortManager.add(cohort3)
+    cohortManager.add(cohort4)
+    cohortManager.add(cohort5)
+
+    const expected = Error('this cohort do not exist')
+    // evaluate
+    const result = cohortManager.remove(6)
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('returns error because of duplicate cohort', () => {
+    //   setup
+    cohortManager.add(cohort3)
+    cohortManager.add(cohort4)
+    cohortManager.add(cohort5)
+
+    const expected = Error('Empty name or name already exist')
+    // evaluate
+    const result = cohortManager.add(cohort5)
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('returns an error for adding cohort without name', () => {
+    //   setup
+    const expected = Error('Empty name or name already exist')
+    // evaluate
+    const result = cohortManager.add(cohortX)
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('search a student by name', () => {
+    //   setup
+    cohortManager.add(cohort5)
+    cohortManager.add(cohort4)
+    cohortManager.add(cohort3)
+
+    cohort5.add(arisaSigrist)
+    cohort4.add(bobRoss)
+    cohort3.add(michelleObama)
+
+    const expected = 'Arisa Sigrist is in Cohort 5'
+    // evaluate
+    const result = cohortManager.searchStudent('Arisa', 'Sigrist')
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('returns error for searching a non-existing student', () => {
+    //   setup
+    cohortManager.add(cohort5)
+    cohortManager.add(cohort4)
+    cohortManager.add(cohort3)
+
+    cohort5.add(arisaSigrist)
+    cohort5.add(bobRoss)
+    cohort5.add(michelleObama)
+
+    const expected = Error('Ed Sheeran do not exist')
+    // evaluate
+    const result = cohortManager.searchStudent('Ed', 'Sheeran')
+    // verify
+    expect(result).toEqual(expected)
+  })
+
+  it('returns no overlapping students', () => {
+    //   setup
+    cohortManager.add(cohort5)
+    cohortManager.add(cohort4)
+    cohortManager.add(cohort3)
+
+    cohort5.add(arisaSigrist)
+    cohort5.add(bobRoss)
+    cohort5.add(michelleObama)
```

Comment:
there is a lot of repetition of these 6 lines, would it be possible to remove the repetition by grouping these tests and applying beforeEach()?